src/js/quick-hull-3d/face.js

```diff
@@ -0,0 +1,270 @@
+var Face = function() {
+    this.mark = this.constructor.VISIBLE;
+};
+
+Face.VISIBLE = 1;
+Face.NON_CONVEX = 2;
+Face.DELETED = 3;
+
+Face.prototype.computeCentroid = function() {
+    var halfEdge = this.halfEdge0;
+    this.centroid = List.zerovector(CSNumber.real(3));
+
+    do {
+        this.centroid = List.add(this.centroid, halfEdge.head().point);
+        halfEdge = halfEdge.next;
+    } while (halfEdge !== this.halfEdge0);
+
+    this.centroid = List.scaldiv(CSNumber.real(this.numberOfVertices), this.centroid);
+};
+
+Face.prototype.computeNormal = function(minArea) {
```

Comment:
There were two version of ```computeNormal```: ```computeNormal(normal)``` and ```computeNormal(normal, minArea)```. Now it is ```computeNormal(minArea)``` (I hope to split it later).
And since instead of Vector I am using List, which operations actually returns a new object, I save normal in this normal. It seems to be correct since in java files to compute normal ```this.normal``` was used: https://github.com/CindyJS/CindyJS/blob/master/src/java/cindyjs/quickhull3d/Face.java#L62. I hope that it is an acceptable change of the API.